src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -123,13 +134,51 @@ public class BlobFileStore : IFileStore
         return fileProperties;
     }
 
+
     private BlockBlobClient GetInstanceBlockBlob(VersionedInstanceIdentifier versionedInstanceIdentifier)
     {
-        string blobName = $"{versionedInstanceIdentifier.StudyInstanceUid}/{versionedInstanceIdentifier.SeriesInstanceUid}/{versionedInstanceIdentifier.SopInstanceUid}_{versionedInstanceIdentifier.Version}.dcm";
+        string blobName;
+        if (_blobMigrationFormatType == BlobMigrationFormatType.New)
+        {
+            blobName = _nameWithPrefix.GetInstanceFileName(versionedInstanceIdentifier);
+        }
+        else
+        {
+            blobName = _nameWithUid.GetInstanceFileName(versionedInstanceIdentifier);
+        }
 
         return _container.GetBlockBlobClient(blobName);
     }
 
+    // TODO: This should removed once we migrate everything and the global flag is turned on
+    private BlockBlobClient[] GetInstanceBlockBlobClients(VersionedInstanceIdentifier versionedInstanceIdentifier)
+    {
+        var clients = new List<BlockBlobClient>();
```

Comment:
Since we know the list can at most contain 2 elements, we should initialize it with that capacity. Also, not sure if there's a noticeable performance hit from creating a list and converting it to an array, vs just using an array here.

Reply:
sure will set the capacity, also I believe there should not be performance degradation for 2 elements